STEP_Project/OUListContent.js

```diff
@@ -173,7 +237,9 @@ function addButtonsToRows(data)
             //hide unnecessary buttons
             if (row.lastChild !== null)
             {
+                restoreOpacity = row.lastChild.style.opacity;
                 row.lastChild.style.opacity = 0;
+                restoreEvents = row.lastChild.style.pointerEvents;
                 row.lastChild.style.pointerEvents = 'none';
```

Comment:
This piece of code is inside a loop. Changing global variable here means that only last iteration of the loop will be retained in this global variable. Similarly for when we are restoring using this global variable. This brings up some obvious questions, do all rows have the same opacity and pointerEvents ? If no, then this won't work and we need to have an array of these. If yes, then we shouldn't set this inside a for loop and instead set it explicitly to one of the row's attributes outside the loop. Also I don't think maintaining this global state is a very nice approach, we should just pass it around in each function around in each function and in listeners.

Reply:
One approach could be, you can also copy the original style into data attribute for each node and restore it later.

`row.lastChild.setAttribute('data-style',  row.getAttribute('style'));` for copying

`row.lastChild.setAttribute('style', row.getAttribute('data-style'));` for restoring

With this approach you won't have to deal with what styles are you changing

Reply:
That's a really good idea, I implemented it. 
Now there's no more global variables, and the style is set separately for each row and restored separately for each row. Although all rows have the same style so far, I've maintained the for loop for adaptability.